Content.Client/UserInterface/OptionsMenu.Audio.cs

```diff
@@ -1,20 +1,156 @@
+                    {
+                        new Control {CustomMinimumSize = (4, 0)},
+                        new Label {Text = Loc.GetString("Master Volume:")},
+                        new Control {CustomMinimumSize = (8, 0)},
+                        MasterVolumeSlider,
+                        new Control {CustomMinimumSize = (8, 0)},
+                        MasterVolumeLabel,
+                        new Control { CustomMinimumSize = (4, 0) },
+                    }
+                });
+
+                ApplyButton = new Button
+                {
+                    Text = Loc.GetString("Apply"), TextAlign = Label.AlignMode.Center,
+                    SizeFlagsHorizontal = SizeFlags.ShrinkEnd
+                };
+
+                vBox.AddChild(new Label
+                {
+                    Text = Loc.GetString("Volume Sliders"),
+                    FontColorOverride = StyleNano.NanoGold,
+                    StyleClasses = { StyleNano.StyleClassLabelKeyText }
+                });
+
+                vBox.AddChild(new MarginContainer
+                {
+                    MarginLeftOverride = 2,
+                    MarginTopOverride = 2,
+                    MarginRightOverride = 2,
+                    SizeFlagsVertical = SizeFlags.FillExpand,
+                    Children =
+                    {
+                        contents
+                    }
+                });
+
+                ResetButton = new Button
                 {
-                    PlaceholderText = "Pretend there's a bunch of volume sliders here."
+                    Text = Loc.GetString("Reset all"),
+                    StyleClasses = { StyleBase.ButtonCaution },
+                    SizeFlagsHorizontal = SizeFlags.ShrinkEnd
+                };
+
+                vBox.AddChild(new StripeBack
+                {
+                    HasBottomEdge = false,
+                    HasMargins = false,
+                    Children =
+                    {
+                        new HBoxContainer
+                        {
+                            Align = BoxContainer.AlignMode.End,
+                            SizeFlagsHorizontal = SizeFlags.FillExpand,
+                            SizeFlagsVertical = SizeFlags.FillExpand,
+                            Children =
+                            {
+                                ResetButton,
+                                new Control { CustomMinimumSize = (2, 0) },
+                                ApplyButton
+                            }
+                        }
+                    }
                 });
+
+                MasterVolumeSlider.Value = _cfg.GetCVar(CVars.AudioMasterVolume) * 100.0f;
+                MasterVolumeLabel.Text = string.Format("{0:0}%", MasterVolumeSlider.Value);
+
+                ApplyButton.OnPressed += OnApplyButtonPressed;
+                ResetButton.OnPressed += OnResetButtonPressed;
+                MasterVolumeSlider.OnValueChanged += OnMasterVolumeSliderChanged;
```

Comment:
Will need to unsubscribe from these as the control is remade every time the round is restarted.